sdk/forms/src/main/java/com/klaviyo/forms/presentation/KlaviyoFormsOverlayActivity.kt

```diff
@@ -0,0 +1,36 @@
+package com.klaviyo.forms.presentation
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.klaviyo.core.Registry
+import com.klaviyo.forms.webview.WebViewClient
+
+/**
+ * Presented over the host application to display a Klaviyo form when triggered.
+ *
+ * TODO On back button, close the form within the webview first (for the css animation)
+ */
+internal class KlaviyoFormsOverlayActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Registry.get<WebViewClient>().attachWebView(this)
+    }
+
+    override fun finish() {
+        Registry.get<WebViewClient>().detachWebView(this)
+        super.finish()
+    }
```

Comment:
This activity will be a little hole in our test coverage, since unit testing these is nigh impossible. So keeping them skinny and testing the services it relies on instead will at least improve confidence.

Reply:
Made this about as skinny as possible by moving more logic to lifecycle listeners in the presentation manager instead!